tests/dragonfly/cluster_test.py

```diff
@@ -2982,6 +2982,64 @@ async def test_cluster_sharded_pub_sub(df_factory: DflyInstanceFactory):
     assert message == {"type": "unsubscribe", "pattern": None, "channel": b"kostas", "data": 0}
 
 
+@dfly_args({"proactor_threads": 2, "cluster_mode": "yes"})
+async def test_cluster_sharded_pubsub_shard_commands(df_factory: DflyInstanceFactory):
+    nodes = [df_factory.create(port=next(next_port)) for i in range(2)]
+    df_factory.start_all(nodes)
+
+    c_nodes = [node.client() for node in nodes]
+
+    nodes_info = [(await create_node_info(instance)) for instance in nodes]
+    nodes_info[0].slots = [(0, 16383)]
+    nodes_info[1].slots = []
+
+    await push_config(json.dumps(generate_config(nodes_info)), [node.client for node in nodes_info])
+
+    node_a = ClusterNode("localhost", nodes[0].port)
+    node_b = ClusterNode("localhost", nodes[1].port)
+
+    consumer_client = RedisCluster(startup_nodes=[node_a, node_b])
+    consumer = consumer_client.pubsub()
+    consumer.ssubscribe("pubsub-shard-channel")
+
+    await c_nodes[0].execute_command("SPUBLISH pubsub-shard-channel 1")
+
+    # Consume subscription message result from above
+    message = consumer.get_sharded_message(target_node=node_a)
+    assert message == {
+        "type": "subscribe",
+        "pattern": None,
+        "channel": b"pubsub-shard-channel",
+        "data": 1,
+    }
+
+    message = await c_nodes[0].execute_command("PUBSUB SHARDCHANNELS")
```

Comment:
nit: consider adding a few more channels, then check that they are printed here. Use a filter below, to check that we do actually filter and apply pattern correctly (which existed before and it should be correct but why not :smile: )